src/containers/ViewerPanel/index.tsx

```diff
@@ -431,6 +462,62 @@ class ViewerPanel extends React.Component<ViewerPanelProps, ViewerPanelState> {
         }
     };
 
+    private updateDisplayBreakpoints = () => {
+        if (location.pathname === EMBED_PATHNAME) {
+            this.setState({
+                embedDisplayBreakpoints: {
+                    belowControlsHeight:
+                        window.innerHeight <=
+                        EMBEDDED_STANDARD_CONTROLS_MINIMUM_HEIGHT,
+                    belowControlsWidth:
+                        window.innerWidth <=
+                        EMBEDDED_STANDARD_CONTROLS_MINIMUM_WIDTH,
+                    belowScaleBarWidth:
+                        window.innerWidth <= EMBEDDED_SCALE_BAR_MINIMUM_WIDTH,
+                },
+            });
+        }
+    };
+
+    private handleFullscreenChange = () => {
+        const { setEmbedFullscreen } = this.props;
+        setEmbedFullscreen(!!document.fullscreenElement);
+        const current = this.centerContent.current;
+        if (current) {
+            setTimeout(() => {
+                this.resize(current);
+                this.updateDisplayBreakpoints();
+            }, 100);
+        }
+    };
+
+    private toggleFullscreen = () => {
+        const { embedFullscreen } = this.props;
+        if (embedFullscreen) {
+            document.exitFullscreen();
+        } else {
+            document.documentElement.requestFullscreen().catch((err) => {
+                console.error(
+                    `Error attempting to enable fullscreen: ${err.message}`
+                );
+            });
+        }
+    };
+
+    private get embedDisplaySettings() {
+        const { belowControlsHeight, belowControlsWidth, belowScaleBarWidth } =
+            this.state.embedDisplayBreakpoints;
+        return {
+            minimalPlaybackControls: belowControlsWidth,
+            minimalCameraControls: belowControlsHeight,
+            onlyRenderBottomControls: belowControlsWidth && belowControlsHeight,
+            showScaleBar:
+                !belowScaleBarWidth &&
+                (!belowControlsWidth || !belowControlsHeight),
+            showCameraControls: !belowControlsWidth || !belowControlsHeight,
+        };
+    }
+
```

Comment:
I debated lots of different ways of handling these settings. Didn't put them in state directly because they would be derived state, although that could reduce lines.

Didn't put the breakpoint state in Redux because only one of the breakpoint values is currently accessed in one another container and the lower components would still need them provided by this container.